### PR TITLE
Option to add X-Delete-At and remote_delete_after.

### PR DESCRIPTION
**Pull request by timuralp**
Optionally, add the specified expiration time to the object's set
expiration date. Applies only to Swift.

### CODE REVIEW

File: s3_sync/sync_swift.py [hunk 1]
```diff
@@ -920,14 +930,32 @@ def _is_meta_synced(self, local_metadata, remote_metadata):
         local_keys = [key.lower() for key in local_metadata.keys()
                       if key.lower().startswith(SWIFT_USER_META_PREFIX) or
                       key.lower() in self.propagated_headers]
+
+        # Add the expiration policy value to the object's expiration date
+        if self.remote_delete_after_offset and 'x-delete-at' in local_keys:
+            local_keys['x-delete-at'] += self.remote_delete_after
```
Comment (mdegerne) [inline s3_sync/sync_swift.py]: Is this valid? I don't see where the header is converted from a string before the call at line 416, for example.

Comment (timuralp) [reply s3_sync/sync_swift.py]: Good catch... working on it.

File: test/integration/test_s3_sync.py [hunk 2]
```diff
@@ -366,6 +366,36 @@ def test_swift_propagate_expiration(self):
...
+        hdrs = self.local_swift('head_object', mapping['container'], obj)
+        self.assertIn('x-delete-at', hdrs.keys())
+        expected_exp = int(hdrs['x-delete-at']) + 3600
+        crawler.run_once()
+        remote_hdrs = self.remote_swift('head_object', mapping['aws_bucket'],
+                                        obj)
+        self.assertIn('x-delete-at', remote_hdrs.keys())
+        self.assertEqual(expected_exp, int(remote_hdrs['x-delete-at']))
+
+        # re-uploading should be a NOOP
```
[the remark this answers is not in the document]
Comment (timuralp) [reply test/integration/test_s3_sync.py]: Yep, it does.

File: s3_sync/sync_swift.py [hunk 3]
```diff
@@ -48,15 +48,22 @@ def __init__(self, *args, **kwargs):
...
-                'propagate_expiration'):
+
+        # Allows an additional offset to be added to the propagated x-delete-at
+        # header value.
+        self.propagate_expiration_offset = self.settings.get(
+            'propagate_expiration_offset', 0)
+
+        if self.settings.get('propagate_expiration') and (
+                not self.remote_delete_after or
+                self.propagate_expiration_offset):
```
[the remark this answers is not in the document]
Comment (timuralp) [reply s3_sync/sync_swift.py]: Yea, let's punt this to the next commit.

[the remark this answers is not in the document]
Comment (thiagodasilva) [reply s3_sync/sync_swift.py]: sounds good

File: test/integration/test_s3_sync.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thiagodasilva) [reply test/integration/test_s3_sync.py]: fwiw, it could also be updated in a follow-up, i tested it manually and it works as described :)

**Pull request merged**